File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        string ans = "";
        stack<char> st;
        for(int i = 0 ; i<s.size() ; i++){
            if(st.empty())
                st.push(s[i]);
            
            else if(s[i]==st.top())
                st.pop();
            else
                st.push(s[i]);
        }
        
        while(!st.empty()){
            ans.push_back(st.top());
            st.pop();
        }
        reverse(ans.begin() , ans.end());
        return ans;
    }
};

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.js
/**
 * @param {string} s
 * @return {string}
 */
var removeDuplicates = function(s) {
    let ans = [];
    for(let i = 0 ; i<s.length ; i++){
        if(s[i]!=ans[ans.length - 1])
            ans.push(s[i]);
        else
            ans.pop();
    }
    return ans.join("");
    
};